Source/WarpX.H

```diff
@@ -49,6 +49,27 @@ enum struct PatchType : int
     coarse
 };
 
+struct ParserWrapper
```

Comment:
Could you add a description of this struct, as well as of its members?
Sorry, I did not make a suggestion because I am not sure what its purpose is, e.g., why not using `GpuParser` directly?

Reply:
I have added a brief description of the struct mentioning that it is intended to safely use the GpuParser. 
The GpuParser class does not have an explicit destructor and the wrapper allows for its safe use avoiding memory leak.

Reply:
Thanks for the description!
Actually since this is a separate struct, could we put this in `Source/Parser/ParserWrapper.H` and include it wherever needed? Or am I missing a complication here?